src/lib.rs

```diff
@@ -182,6 +183,55 @@ impl Pixels {
         PixelsBuilder::new(width, height, surface_texture).build()
     }
 
+    /// Resize the pixel buffer, this doesn't resize the surface upon which the pixel buffer is rendered.
+    pub fn resize_buffer(&mut self, width: u32, height: u32) {
+        self.context.texture_extent = wgpu::Extent3d {
+            width,
+            height,
+            depth: 1,
+        };
+
+        // Update ScalingMatrix for mouse transformation
+        self.scaling_matrix_inverse = renderers::ScalingMatrix::new(
+            (
+                self.context.texture_extent.width as f32,
+                self.context.texture_extent.height as f32,
+            ),
+            (width as f32, height as f32),
+        )
```

Comment:
Hm, this looks suspicious. It's passing the same size for both the backing texture (first tuple) and surface texture (second tuple). I believe it should be this:

```suggestion
        self.scaling_matrix_inverse = renderers::ScalingMatrix::new(
            (width as f32, height as f32),
            (
                self.surface_size.width as f32,
                self.surface_size.height as f32,
            ),
        )
```

Reply:
We are in function `resize()` which is used to resize the surface when the window is resized. So, `width` and `height` are the dimensions of the surface, not the backing texture.
And indeed, 2 lines earlier we have:

        // Update SurfaceTexture dimensions
        self.surface_size.width = width;
        self.surface_size.height = height;

Unless I'm missing something that means both tuples of the matrix function would be the same in your proposal.

Reply:
Good catch, updated.